src/components/Dropzone/Dropzone.vue

```diff
@@ -0,0 +1,345 @@
+        class="mt-2 text-gray-700 text-default-light"
+      >
+        {{ textContent.or }}
+      </div>
+
+      <input
+        :id="uploadId"
+        ref="fileElement"
+        type="file"
+        :accept="acceptType"
+        class="hidden"
+        @change="onSelectFile"
+      >
+      <label
+        v-if="defaultStep || errorStep"
+        :for="uploadId"
+        role="button"
+        tabindex="0"
+        :class="['mt-4 flex items-center rounded-lg px-4 h-[32px]',
+                 'primaryBtnStyle text-white font-medium text-sm',
+                 'focus-visible:ring-4 focus-visible:ring-primary-100 focus:outline-none']"
+        @click.stop
+      >
+        {{ textContent.uploadFileButtonText }} <Upload class="w-5 h-5 ml-2 -mr-2" />
+      </label>
+
+      <button
+        v-if="successStep"
+        type="reset"
+        aria-label="Remove file"
+        class="mt-6 underline text-gray-500 !text-sm"
+        @click.stop="removeFile"
+      >
+        {{ textContent.removeFileButtonText }}
+      </button>
+
+      <ProgressBar
+        v-if="uploadingStep"
+        :percentage="progress"
+        class="mt-6"
+      />
+    </form>
+    <div class="mt-2 flex items-center justify-between">
+      <div
+        v-if="showTypeAndMaxSize"
+        :class="['text-sm', errorStep ? 'text-error' : 'text-gray-300']"
+      >
+        <span v-if="acceptType && acceptType !== '/*'">
+          <span v-if="fileTypesArray.length === 1">
+            {{ textContent.acceptedFormatIs }} <strong>{{ fileTypesString }}</strong>.
+          </span>
+          <span v-if="fileTypesArray.length > 1">
+            {{ textContent.acceptedFormatsAre }} <strong>{{ fileTypesString }}</strong>.
+          </span>
+        </span>
+        <span v-if="maxSizeInBytes">
+          {{ textContent.maxFileSizeIs }} <strong>{{ formatBytes(maxSizeInBytes) }}</strong>.
+        </span>
+      </div>
+      <LobLink
+        v-if="sampleLinkUrl"
+        :to="sampleLinkUrl"
+        target="_blank"
+        class="!text-sm"
+      >
+        {{ textContent.downloadSampleFile }}
+      </LobLink>
+    </div>
+  </div>
+</template>
+
+<script>
+import { formatBytes } from '@/utils/formatBytes';
+import { LobLink, ProgressBar, Upload } from '@/components';
+
+export default {
+  name: 'Dropzone',
+  components: { LobLink, ProgressBar, Upload },
+  props: {
+    textContent: { type: Object, required: true,
+      validator: function (value) {
+        const requiredKeys = ['dragAndDropHere', 'yourFile', 'or', 'uploadFileButtonText',
+          'acceptedFormatIs', 'acceptedFormatsAre', 'maxFileSizeIs', 'downloadSampleFile',
+          'uploading', 'mightTakeAMinute', 'looksGreat', 'removeFileButtonText',
+          'couldNotUpload', 'canOnlySelectOneFile', 'fileIsTooLarge', 'fileTypeNotValid',
+          'defaultErrorText', 'errorMessage', 'successMessage'];
+        for (const key of requiredKeys) {
+          return (Object.keys(value).includes(key));
```

Comment:
This is not quite what we want:
1. It's attempting to create an array of the `keys` in the `value` for every iteration over the `requiredKeys` rather than just doing that work once (while we're only dealing with a tiny amount data here and it probably is insignificant in the end, creating/modifying arrays does take memory and time so we should always make sure we're not doing it any more than necessary).
2. Because you have a `return` in the loop it's actually only going to ever check the first `key` and then exit, so we're not actually validating all of the `requiredKeys` are in the `value`.

Something you could do instead:
```
return requiredKeys.every((key) => value.hasOwnProperty(key));
```

(see: 
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
)